lgc/patch/PatchInOutImportExport.h

```diff
@@ -222,9 +225,10 @@ class PatchInOutImportExport : public Patch {
   llvm::GlobalVariable *m_lds; // Global variable to model LDS
   llvm::Value *m_threadId;     // Thread ID
 
-  std::vector<llvm::CallInst *> m_importCalls;                 // List of "call" instructions to import inputs
-  std::vector<llvm::CallInst *> m_exportCalls;                 // List of "call" instructions to export outputs
-  PipelineState *m_pipelineState = nullptr;                    // Pipeline state from PipelineStateWrapper pass
+  std::vector<llvm::CallInst *> m_importCalls;          // List of "call" instructions to import inputs
+  std::vector<llvm::CallInst *> m_exportCalls;          // List of "call" instructions to export outputs
+  std::map<unsigned, llvm::Value *[4]> m_attribExports; // Export info of vertex attributes
```

Comment:
What is does `m_attribExports` map from? IE what does the `unsigned` key represent, what values are valid, etc.

Reply:
m_attribExports is from the collecting of potential exp instruction (exp instruction has such form: `exp paramN, val0, val1, val2, val3`). The value is <attribute location, attribute values(4 x float)>. I will add more explanation comments.